components/bookshelf/detailedbook/utils.ts

```diff
@@ -0,0 +1,22 @@
+import { BookType } from "api/types"
```

Comment:
`utils.ts` should always be in a `shared` folder.

can you put this to `/utils.ts` from root or `/utils/detailedBook`?

You can also try this method for the shared functions above.